src/usr/bin/diffuse

```diff
@@ -25,6 +25,9 @@ import gettext
 import locale
 import os
 import sys
+import warnings
+
+warnings.filterwarnings("ignore")
```

Comment:
Well, I don't think this change is what we want. As a developer I want to see all the warnings, and I want the users to be able to send me the warning list in case of a bug. Is there a reason other than aesthetic for this change?

Reply:
As I user I did not want to see them:
```
/usr/local/bin/diffuse:7472: DeprecationWarning: Gtk.IconSet.new_from_pixbuf is deprecated
  factory.add(DIFFUSE_STOCK_NEW_2WAY_MERGE, Gtk.IconSet.new_from_pixbuf(p))
/usr/local/bin/diffuse:7472: DeprecationWarning: Gtk.IconFactory.add is deprecated
  factory.add(DIFFUSE_STOCK_NEW_2WAY_MERGE, Gtk.IconSet.new_from_pixbuf(p))
/usr/local/bin/diffuse:7508: DeprecationWarning: Gtk.IconFactory.add_default is deprecated
  factory.add_default()
/usr/local/bin/diffuse:1180: DeprecationWarning: Gtk.ImageMenuItem.new_with_mnemonic is deprecated
  item = Gtk.ImageMenuItem.new_with_mnemonic(spec[0])
/usr/local/bin/diffuse:1187: DeprecationWarning: Gtk.Image.set_from_stock is deprecated
  image.set_from_stock(spec[3], Gtk.IconSize.MENU)
/usr/local/bin/diffuse:1188: DeprecationWarning: Gtk.ImageMenuItem.set_image is deprecated
  item.set_image(image)
/usr/local/bin/diffuse:7618: DeprecationWarning: Gtk.HBox.new is deprecated
  hbox = Gtk.HBox.new(False, 0)
/usr/local/bin/diffuse:1235: DeprecationWarning: Gtk.VSeparator.new is deprecated
  separator = Gtk.VSeparator.new()
/usr/local/bin/diffuse:6832: DeprecationWarning: Gtk.Misc.set_alignment is deprecated
  label.set_alignment(0, 0.5)
/usr/local/bin/diffuse:2717: DeprecationWarning: Gtk.VScrollbar.new is deprecated
  self.vbar = bar = Gtk.VScrollbar.new(vadj)
/usr/local/bin/diffuse:2721: DeprecationWarning: Gtk.HScrollbar.new is deprecated
  self.hbar = bar = Gtk.HScrollbar.new(hadj)
```

Reply:
I believe that ignoring warnings should not be the default behaviour of a program, but a choice of the person that starts it (that's the reason why we can ignore them when starting a script, with `-W ignore`).
Maybe we can set this in launcher file or have this choice passed as a parameter to the program?

That said, if you decide to really go through with it, then might I suggest to use 
```suggestion
warnings.filterwarnings("ignore", category=DeprecationWarning)
```
so that only the deprecations are ignored and other warnings are shown?

Reply:
Sorry, that was me with an old account that I had to delete. Just re-replying, I am not a ghost.

Reply:
I was getting scared there for a sec! ;)

I'll remove it from the PR and keep it in my local version.

Reply:
Lots of GUI applications have warnings what started from the command line, but lots of GUI applications are not launched through the command line either.

Our role as developers is to fix these console-only warnings, not hide them because we think it is not pleasing to the eye. We already did a pretty good pass on them, the ones left are mostly due to issues with gtk3 python libs... When I tried to fix them, Diffuse wasn't working anymore, so I'll just wait for another release of the gtk libs and see if it works.